Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/WANDPowderReductionTest.py

```diff
@@ -351,37 +364,86 @@ def test_event(self):
             Function="Flat background",
         )
 
+        #CASE 1
+        #input single workspace, output single workspace
         pd_out = WANDPowderReduction(
             InputWorkspace=event_data,
             CalibrationWorkspace=event_cal,
             BackgroundWorkspace=event_bkg,
             Target="Theta",
             NumberBins=1000,
             NormaliseBy="None",
+            Sum=False,
         )
 
-        x = pd_out.extractX()
-        y = pd_out.extractY()
+        print(pd_out.getNumberOfEntries())
 
-        self.assertAlmostEqual(x.min(), 0.03517355)
-        self.assertAlmostEqual(x.max(), 70.3119282)
-        self.assertAlmostEqual(y[0, 0], 0.0)
+        for i in pd_out:
+
+            x = i.extractX()
+            y = i.extractY()
+            
+            print(x.min(), x.max())
 
+            self.assertAlmostEqual(x.min(), 0.03517355)
+            self.assertAlmostEqual(x.max(), 70.3119282)
+            self.assertAlmostEqual(y[0, 0], 0.0)
+
+        assert isinstance(pd_out, WorkspaceGroup)
+        assert len(pd_out) == 1
+
+        #CASE 2
+        #input multiple single ws, output (single) summed ws
         pd_out = WANDPowderReduction(
             InputWorkspace=[event_data, event_data],
             CalibrationWorkspace=event_cal,
-            BackgroundWorkspace="event_bkg,event_bkg",
+            BackgroundWorkspace=event_bkg,
             Target="Theta",
             NumberBins=1000,
             NormaliseBy="None",
+            Sum=True,
         )
-
+        
         x = pd_out.extractX()
         y = pd_out.extractY()
 
         self.assertAlmostEqual(x.min(), 0.03517355)
         self.assertAlmostEqual(x.max(), 70.3119282)
         self.assertAlmostEqual(y[0, 0], 0.0)
+        assert isinstance(pd_out, MatrixWorkspace)
+        
+        #CASE 3
+        #DOES NOT WORK YET -- currently outputs Workspace2D -- will modify when algorithm is updated
+        #input group ws containing several ws, output group ws containing several ws
+        groupWS = WorkspaceGroup()
```

Comment:
Pass in the list(str) of the workspace names rather than a workspace group. The algorithm doesn't know about workspace groups how it is written